ci.cue

```diff
@@ -51,19 +52,25 @@ dagger.#Plan & {
 			output: _revision.export.files["/revision"]
 		}
 
-		build: go.#Build & {
-			source:  _source
-			package: "./cmd/dagger/"
-			os:      client.platform.os
-			arch:    client.platform.arch
+		build: {
+			"go": go.#Build & {
+				source:  _source
+				package: "./cmd/dagger/"
+				os:      client.platform.os
+				arch:    client.platform.arch
 
-			ldflags: "-s -w -X go.dagger.io/dagger/version.Revision=\(version.output)"
+				ldflags: "-s -w -X go.dagger.io/dagger/version.Revision=\(version.output)"
 
-			env: {
-				CGO_ENABLED: "0"
-				// Makes sure the linter and unit tests complete before starting the build
-				// "__depends_lint":  "\(goLint.exit)"
-				// "__depends_tests": "\(goTest.exit)"
+				env: {
+					CGO_ENABLED: "0"
+					// Makes sure the linter and unit tests complete before starting the build
+					// "__depends_lint":  "\(goLint.exit)"
+					// "__depends_tests": "\(goTest.exit)"
+				}
+			}
+			docker: core.#Dockerfile & {
+				source: _source
+				dockerfile: path: "Dockerfile"
```

Comment:
Can we ignore the `website` folder here? it's a big one and we're sending it to the context unnecessarily

Reply:
Have to hack around `filesystem` as it seems we can't inherit from another one so I have created a filesystem to build the docker image named `./`. This way it doesn't overlap with the main one called `.` but that looks hacky to me.

before

```
11:47AM INF actions.build.docker | completed    duration=2m1.1s
```

after

```
3:53PM INF actions.build.docker | completed    duration=1m45.9s
```

Would be better if we could specify an `id` for a filesystem or better inherit from an existing one to override some attributes.

Reply:
> You can use `docker.#Copy` to create a new `FS` with the things you want to include or exclude. I think doing a `copy` vs creating a new FS should be more performant

iiuc it will build two images? First one that will just create an FS with just a `docker.#Copy` step and the `core.#Dockerfile` will use this file to build its image. That sounds worse no?

Maybe we should discuss and make this change in a follow-up as it seems out of scope with this PR. WDYT?

Reply:
SGTM!

Reply:
> iiuc it will build two images? First one that will just create an FS with just a `docker.#Copy` step and the `core.#Dockerfile` will use this FS to build its image? That sounds worse no?

Correct. I meant `core.#Copy` which don't produce an `#Image` but a new `dagger.#FS`